### PR TITLE
api boomark shows 1962

### LINKED ISSUE #1962: Catalog:  Filter 'My APIs / Bookmarks' from 'Next' page not working
# Steps  
1. Visit https://nightly.apinf.io  
2. Sign in 
3. Add APIs 
4. Go to Catalog  (default is 'Show all' )
5. Select and bookmark any API   
6. Go to next page by clicking on 'Next' / page 2   
7.  Click on My APIs/ My Bookmarks  

# Expected Result
Displays all 'My APIs /My Bookmarks' successfully   

# Actual Result
Shows an empty page   

# Screenshot   
catalog_next_myapis_empty_page

catalog_nextpage_mybookmarks

### PR DESCRIPTION
**Pull request by snittoor**
Closes #1962 

Apibookmarks where not showing . This is because instance.subscription('myApibookmarks') was not  ready. This instance.subscription is now initialized in cataloge.js (parent) instead of bookmark.js

### CODE REVIEW

**Comment by jykae** [comment]
Reviewing

File: bookmarks/client/bookmark.js [hunk 1]
```diff
@@ -24,25 +24,30 @@ Template.bookmark.events({
...
 
     // Get reference to template instance
     const instance = Template.instance();
 
-    // Make sure bookmark subscription is ready
-    if (instance.bookmarksSubscription.ready()) {
-      // Get current user bookmark (should be only one API Bookmarks result available)
-      const userBookmarks = ApiBookmarks.findOne({ userId: Meteor.user()._id, apiIds: apiId });
+    // Get API ID from instance data context
+    const apiId = instance.api._id;
```
Comment (jykae) [inline bookmarks/client/bookmark.js]: @snittoor @brylie this line returns undefined. To get template instance you could use instance.data.api._id or Template.currentData() Personally I prefer the latter as it's straight access to dynamic data context without using "this".

File: bookmarks/client/bookmark.js [hunk 2]
```diff
@@ -24,25 +24,30 @@ Template.bookmark.events({
...
 
-      // Make sure user has bookmarks
-      if (userBookmarks) {
-        return true;
-      }
+    // Get current user bookmarks, only if this API is bookmarked
+    const userBookmarks = ApiBookmarks.findOne({
+      userId: Meteor.user()._id,
+      apiIds: apiId,
```
Comment (jykae) [inline bookmarks/client/bookmark.js]: Also make sure to check case "undefined" by using if (apiId) structure before using apiId.

File: catalogue/client/catalogue.js [hunk 3]
```diff
@@ -6,16 +6,28 @@ import { FlowRouter } from 'meteor/kadira:flow-router';
 import { Apis } from '/apis/collection';
 import { ApiBookmarks } from '/bookmarks/collection';
 
+import { $ } from 'jquery';
```
Comment (jykae) [inline catalogue/client/catalogue.js]: This import is incorrect and results error later on code. 
```
TypeError: $ is not a function
    at Blaze.TemplateInstance.<anonymous> (catalogue.js:127)
    at blaze.js?hash=891d5a1…:3341
    at Function.Template._withTemplateInstanceFunc (blaze.js?hash=891d5a1…:3687)
    at fireCallbacks (blaze.js?hash=891d5a1…:3337)
    at Blaze.View.<anonymous> (blaze.js?hash=891d5a1…:3430)
    at blaze.js?hash=891d5a1…:1783
    at Object.Blaze._withCurrentView (blaze.js?hash=891d5a1…:2214)
    at blaze.js?hash=891d5a1…:1782
    at Object.Tracker._runFlush (tracker.js?hash=e52e5fe…:539)
    at onGlobalMessage (meteor.js?hash=f9ccb2f…:401)
```
Changing to "import $ from jquery" should fix issue. Or possibly better to use local jquery http://blazejs.org/api/templates.html#Blaze-TemplateInstance-\$
eg. 
```
Template.catalogue.onRendered(function () {
  // Activate tooltips on all relevant items
  this.$('.toolbar-tooltip').tooltip({ placement: 'bottom' });
});
```
Here this points to Template instance, you could also save it to variable to make it more clear, "const instance = this", and then use "instance.$"

This is safer, efficient & does not require importing jquery.
So import can be removed if you decide to use local jquery on line 127.

**Comment by jykae** [comment]
@snittoor @brylie Overall 2 small errors, provided suggestions for fixes. Keep up good work 👍 @snittoor welcome to community!

File: bookmarks/client/bookmark.js (hunk 2 above)
Comment (snittoor) [inline bookmarks/client/bookmark.js]: Yes even i got it while testing , but didnt know how to fix it .. will try what you have suggested.

File: catalogue/client/catalogue.js (hunk 3 above)
Comment (snittoor) [inline catalogue/client/catalogue.js]: Hi Ville ,
  Thank you for comments and suggestions. Will implement them , though i did not get the above error will look into this error also.Glad that you tested it.

File: bookmarks/client/bookmark.js (hunk 1 above)
Comment (brylie) [inline bookmarks/client/bookmark.js]: @jykae when does this line (36) return undefined?

```js
// Get API ID from instance data context
const apiId = instance.api._id;
```

File: bookmarks/client/bookmark.js (hunk 2 above)
Comment (brylie) [inline bookmarks/client/bookmark.js]: This is resolved by fixing line 36.

File: catalogue/client/catalogue.js (hunk 3 above)
Comment (brylie) [inline catalogue/client/catalogue.js]: As @jykae mentioned, change line 9 to `import $ from jquery`

Comment (snittoor) [inline catalogue/client/catalogue.js]: @brylie : Done , changed the code . Please review.

File: bookmarks/client/bookmark.js [hunk 4]
```diff
@@ -33,14 +33,14 @@ Template.apiBookmark.helpers({
...
-    const apiId = instance.api._id;
+    const apiId = instance.data.api._id;
 
     // Get current user bookmarks, only if this API is bookmarked
     const userBookmarks = ApiBookmarks.findOne({
       userId: Meteor.user()._id,
       apiIds: apiId,
     });
-
+console.log("User Bookmarks",userBookmarks);
```
Comment (brylie) [inline bookmarks/client/bookmark.js]: Remove console logs. We try not to leave these logging messages.

**Comment by brylie** [comment]
@snittoor this looks great. One last request is that you remove the `console.log()` statement. We try not to have console logs in our app.

File: bookmarks/client/bookmark.js (hunk 4 above)
Comment (snittoor) [inline bookmarks/client/bookmark.js]: Done.

**Pull request merged**